### PR TITLE
Refactor and increase test coverage

### PR DESCRIPTION
**Pull request by rollulus**

### CODE REVIEW

File: pkg/landscaper/component_provider.go [hunk 1]
```diff
@@ -135,8 +135,8 @@ func (cp *fileComponentProvider) get(files []string) (Components, error) {
 		}
 
 		if len(cmp.Secrets) > 0 {
-			sort.Strings(cmp.Secrets) // enforce a consistent ordering for proper diffing / deepEqualing
-			readSecretValues(cmp)
+			sort.Strings(cmp.Secrets)            // enforce a consistent ordering for proper diffing / deepEqualing
+			readSecretValuesFromEnvironment(cmp) // TODO: remove coupling; make secrets source pluggable
```
Comment (johnhofman) [inline pkg/landscaper/component_provider.go]: Can you link this todo to the issue?

File: pkg/landscaper/executor.go [hunk 2]
```diff
@@ -42,8 +48,8 @@ func (e *executor) Apply(desired, current Components) error {
 	needForcedUpdate := map[string]bool{}
 	for _, cmp := range update {
 		// to work around k8s #35149, cronJobs need a force update
-		if e.env.NoCronUpdate {
```
Comment (johnhofman) [inline pkg/landscaper/executor.go]: Nice reduction of breaking the Law of Demeter.

File: pkg/landscaper/executor.go [hunk 3]
```diff
@@ -22,15 +22,21 @@ type Executor interface {
 }
 
 type executor struct {
-	env             *Environment
+	helmClient      helm.Interface
+	chartLoader     ChartLoader
 	secretsProvider SecretsProvider
+	noCronUpdate    bool
```
Comment (johnhofman) [inline pkg/landscaper/executor.go]: These options smell a bit, but I don't know enough Go to know why. I think it is that they are set to `false, false` in all the tests and I'd expect some sort of default value mechanism.

Comment (rollulus) [inline pkg/landscaper/executor.go]: In the tests they bothered me as well. One way would be to have an additional `ExecutorOptions` `struct` with just flags in it, which can be passed to `NewExecutor`. Another approach is what Dave Cheney calls ["functional options"](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis) and this is what Helm does e.g. [here](https://godoc.org/k8s.io/helm/pkg/helm#Client.InstallRelease)

**Pull request merged**